distage/distage-testkit-scalatest/src/main/scala-3/izumi/distage/testkit/scalatest/AssertSync.scala

```diff
@@ -9,6 +10,29 @@ trait AssertSync[F[_]] {
   inline final def assertIO(inline arg: Boolean)(implicit Sync: Sync[F], prettifier: Prettifier, pos: source.Position): F[Assertion] = {
     Sync.delay(Assertions.assert(arg))
   }
+
+  inline final def assertIO[T](
+    inline effect: F[T]
+  )(inline predicate: T => Boolean
+  )(implicit Sync: Sync[F],
+    FlatMap: FlatMap[F],
+    prettifier: Prettifier,
+    pos: source.Position,
+  ): F[Assertion] = {
+    FlatMap.flatMap(effect)(result => assertIO(predicate(result)))
+  }
+
+  inline final def assertIO[A, B](
+    inline effectA: F[A],
+    inline effectB: F[B],
+  )(inline predicate: (A, B) => Boolean
+  )(implicit Sync: Sync[F],
+    FlatMap: FlatMap[F],
```

Comment:
This is redundant because Sync already contains FlatMap
```suggestion
    Sync: Sync[F],
```